examples/Datalogger/Datalogger.ino

```diff
@@ -52,21 +66,13 @@ void loop()
     }
   }
 
-  // open the file. note that only one file can be open at a time,
-  // so you have to close this one before opening another.
-  File dataFile = SD.open("datalog.txt", FILE_WRITE);
 
   // if the file is available, write to it:
   if (dataFile) {
-    dataFile.seek(dataFile.size());
     dataFile.println(dataString);
-    dataFile.close();
+    dataFile.flush(); // use flush to ensure the data written
     // print to the serial port too:
     Serial.println(dataString);
   }
-  // if the file isn't open, pop up an error:
-  else {
-    Serial.println("error opening datalog.txt");
-  }
```

Comment:
Maybe it could be fine to kept the else in case of `dataFile `issue?
```patch
+  // if the file isn't open, pop up an error:
+  else {
+    Serial.println("error on datalog.txt file handle");
+  }
```

Reply:
Since it already handled on `setup()` function and `dataFile` never changed anymore since declared, i think it useless to put it on loop, so i was thinking to remove it, no? But no problem to kept it

Reply:
Right in this example. My concern was for hardening. There is an if so an else to warn if we reach it is fine I guess.  If the ptr is corrupted... even if this should not happen

Reply:
Sorry how to do this? Should i cancel the pull request, make commit then pull request again?

Reply:
Simply push on the same branch then this will update this pr

Reply:
Ok i just pushed the changes